src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -254,9 +255,27 @@ public override SyntaxNode VisitSimpleLambdaExpression(SimpleLambdaExpressionSyn
             public override SyntaxNode VisitArgument(ArgumentSyntax node)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
-
+                string inferredName = null;
+                if (node.Parent.IsKind(SyntaxKind.TupleExpression))
+                {
+                    var tuple = (TupleExpressionSyntax)node.Parent;
+                    var nonDuplicateTupleNames = ExtractTupleNames(tuple);
+                    inferredName = ExtractAnonymousTypeMemberName(node.Expression).ValueText;
+                    if (!nonDuplicateTupleNames.Contains(inferredName))
+                    {
+                        inferredName = null;
+                    }
+                }
                 var newArgument = (ArgumentSyntax)base.VisitArgument(node);
 
+                if (inferredName != null && node.NameColon == null)
+                {
+                    var identifier = SyntaxFactory.Identifier(inferredName);
+                    identifier = TryEscapeIdentifierToken(identifier, node, _semanticModel);
+                    newArgument = newArgument.WithNameColon(SyntaxFactory.NameColon(SyntaxFactory.IdentifierName(identifier)))
```

Comment:
I added a test with trivia, but it seems to be attached to the open paren. I'm not sure how to get trivia attached to the first tuple element.